src/api/procedures/executeManualInstruction.ts

```diff
@@ -0,0 +1,174 @@
+import { PolymeshPrimitivesIdentityIdPortfolioId } from '@polkadot/types/lookup';
+import BigNumber from 'bignumber.js';
+import P from 'bluebird';
+
+import { assertInstructionValidForManualExecution } from '~/api/procedures/utils';
+import { Instruction, PolymeshError, Procedure } from '~/internal';
+import {
+  DefaultPortfolio,
+  ErrorCode,
+  ExecuteManualInstructionParams,
+  InstructionDetails,
+  Leg,
+  NumberedPortfolio,
+  TxTags,
+} from '~/types';
+import { ExtrinsicParams, ProcedureAuthorization, TransactionSpec } from '~/types/internal';
+import {
+  bigNumberToU32,
+  bigNumberToU64,
+  portfolioIdToMeshPortfolioId,
+  portfolioLikeToPortfolioId,
+  u64ToBigNumber,
+} from '~/utils/conversion';
+
+export interface Storage {
+  portfolios: (DefaultPortfolio | NumberedPortfolio)[];
+  totalLegAmount: BigNumber;
+  instructionDetails: InstructionDetails;
+  signer: string;
+}
+
+/**
+ * @hidden
+ */
+export async function prepareExecuteManualInstruction(
+  this: Procedure<ExecuteManualInstructionParams, Instruction, Storage>,
+  args: ExecuteManualInstructionParams
+): Promise<
+  TransactionSpec<Instruction, ExtrinsicParams<'settlementTx', 'executeManualInstruction'>>
+> {
+  const {
+    context: {
+      polymeshApi: {
+        tx: { settlement: settlementTx },
+        query: { settlement },
+      },
+    },
+    context,
+    storage: { portfolios, totalLegAmount, instructionDetails, signer },
+  } = this;
+
+  const { id } = args;
+
+  const instruction = new Instruction({ id }, context);
+
+  await assertInstructionValidForManualExecution(instructionDetails, context);
+
+  if (!portfolios.length) {
+    const {
+      owner: { did: ownerDid },
```

Comment:
```suggestion
      owner: { did: venueOwner },
```
Here I think "owner" implies Identity enough we don't need the "did" part.